src/Parser/Druid/Feral/Modules/Bleeds/MoonfireSnapshot.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Snapshot, { PANDEMIC_FRACTION } from '../FeralCore/Snapshot';
+
+/**
+ * Moonfire benefits from the damage bonus of Tiger's Fury over its whole duration, even if the
+ * buff wears off in that time. It's a damage loss to refresh Moonfire before the pandemic window
+ * if you don't have Tiger's Fury active when the existing DoT does have it active.
+ */
+
+// cat moonfire lasts for 14 seconds, unlike caster and bear moonfire with a base of 16 seconds.
+const MOONFIRE_FERAL_BASE_DURATION = 14000;
+
+class MoonfireSnapshot extends Snapshot {
+  moonfireCastCount = 0;
+  downgradeCastCount = 0;
+
+  on_initialized() {
+    if (!this.combatants.selected.hasTalent(SPELLS.LUNAR_INSPIRATION_TALENT.id)) {
+      this.active = false;
+      return;
+    }
+
+    this.spellCastId = SPELLS.MOONFIRE_FERAL.id;
+    this.debuffId = SPELLS.MOONFIRE_FERAL.id;
+
+    // unlike bleeds, Moonfire's duration is not affected by the Jagged Wounds talent
+    this.durationOfFresh = MOONFIRE_FERAL_BASE_DURATION;
+    this.isProwlAffected = false;
+    this.isTigersFuryAffected = true;
+
+    // bloodtalons only affects melee abilities
+    this.isBloodtalonsAffected = false;
+  }
+
+  on_byPlayer_cast(event) {
+    if (SPELLS.MOONFIRE_FERAL.id === event.ability.guid) {
+      ++this.moonfireCastCount;
+    }
+    super.on_byPlayer_cast(event);
+  }
+
+  checkRefreshRule(stateNew) {
+    const stateOld = stateNew.prev;
+    if (!stateOld || stateOld.expireTime < stateNew.startTime) {
+      // not a refresh, so nothing to check
+      return;
+    }
+    
+    if (stateNew.startTime >= stateOld.pandemicTime ||
+        stateNew.power >= stateOld.power) {
+      // good refresh
+      return;
+    }
+    
+    ++this.downgradeCastCount;
+    
+    // this downgrade is relatively minor, so don't overwrite cast info from elsewhere
```

Comment:
Doesn't look like you're setting it anywhere else so you could leave this part out